plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergNessieCatalogConnectorSmokeTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.containers.NessieContainer;
+import org.apache.iceberg.FileFormat;
+import org.testcontainers.containers.Network;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.testcontainers.containers.Network.newNetwork;
+
+public class TestIcebergNessieCatalogConnectorSmokeTest
+        extends BaseIcebergConnectorSmokeTest
+{
+    private static NessieContainer nessieContainer;
+    private static Network network;
+    private static Path tempDir;
+
+    public TestIcebergNessieCatalogConnectorSmokeTest()
+    {
+        super(FileFormat.PARQUET);
+    }
+
+    @BeforeClass
+    @Override
+    public void init()
+            throws Exception
+    {
+        network = newNetwork();
+        nessieContainer = NessieContainer.builder().withNetwork(network).build();
```

Comment:
wrap with `closeAfterClass` so that there's no extra logic for this in teardown.

Reply:
good point, updated